From e6fb688e1cca192eca7149b673d4b0cec28c4bf1 Mon Sep 17 00:00:00 2001
From: Rajesh Bhagat <rajesh.bhagat@freescale.com>
Date: Mon, 25 Jan 2016 16:11:46 +0530
Subject: [PATCH 359/741] usb: gadget: add a sanity check for complete handler

commit b3347470e5dc7f8ea8bbce48f3660715f0b59289 from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux

complete handler is replaced by usb_gadget_giveback_request by
below commit,Which removed the sanity check which was added
for fsl specific gadget driver.usb: gadget: Refactor request
completion

Signed-off-by: Rajesh Bhagat <rajesh.bhagat@nxp.com>
Signed-off-by: yinbo.zhu <yinbo.zhu@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/usb/gadget/udc/fsl_udc_core.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/usb/gadget/udc/fsl_udc_core.c b/drivers/usb/gadget/udc/fsl_udc_core.c
index 9a05863b2876..db360f04b4aa 100644
--- a/drivers/usb/gadget/udc/fsl_udc_core.c
+++ b/drivers/usb/gadget/udc/fsl_udc_core.c
@@ -194,7 +194,11 @@ __acquires(ep->udc->lock)
 
 	spin_unlock(&ep->udc->lock);
 
-	usb_gadget_giveback_request(&ep->ep, &req->req);
+	/* this complete() should a func implemented by gadget layer,
+	 * eg fsg->bulk_in_complete()
+	 */
+	if (req->req.complete)
+		usb_gadget_giveback_request(&ep->ep, &req->req);
 
 	spin_lock(&ep->udc->lock);
 	ep->stopped = stopped;
-- 
2.17.1

